smartsim/_core/control/jobmanager.py

```diff
@@ -328,20 +324,19 @@ def set_db_hosts(self, orchestrator: Orchestrator) -> None:
         :type orchestrator: Orchestrator
         """
         # should only be called during launch in the controller
-        self._lock.acquire()
-        try:
+        with self._lock:
             if orchestrator.batch:
                 self.db_jobs[orchestrator.name].hosts = orchestrator.hosts
             else:
                 for dbnode in orchestrator.dbnodes:
-                    if not dbnode._mpmd:
+                    if not dbnode.is_mpmd:
```

Comment:
Another case of exposing a protected variable to users

Reply:
ditto on response. If we're already touching it externally, the name should reflect that until a refactor changes our access patterns.